sentry/src/routes/campaign.rs

```diff
@@ -214,6 +215,40 @@ pub async fn create_campaign<A: Adapter>(
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
 
+pub async fn campaign_list<A: Adapter>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let mut query =
+        serde_urlencoded::from_str::<CampaignListQuery>(req.uri().query().unwrap_or(""))?;
+
+    if query.validator.is_none() {
+        let session = req
+            .extensions()
+            .get::<Auth>()
+            .expect("Auth should exist once we're here");
+        query.validator = Some(session.uid);
+    }
```

Comment:
To have none of them is also a valid case (i.e. when user is unauthenticated and there is no validator in the query).
If no query validator is passed, use an `Option<Auth>` - if it's not set we still want to exclude this filter & Auth is not required for this route, see the #377 list of routes